applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp

```diff
@@ -343,9 +329,135 @@ double ComputeHessianSolMetricProcess<TDim, TVarType>::CalculateAnisotropicRatio
+        for(int i = 0; i < num_nodes; ++i) {
+            auto it_node = nodes_array.begin() + i;
+            it_node->SetValue(tensor_variable, aux_zero_vector);
+        }
+    }
+
+    // Ratio reference variable
+    KRATOS_ERROR_IF_NOT(KratosComponents<Variable<double>>::Has(mRatioReferenceVariable)) << "Variable " << mRatioReferenceVariable << " is not a double variable" << std::endl;
+    const auto& reference_var = KratosComponents<Variable<double>>::Get(mRatioReferenceVariable);
+
+    #pragma omp parallel for
+    for(int i = 0; i < num_nodes; ++i) {
+        auto it_node = nodes_array.begin() + i;
+
+        const Vector& hessian = it_node->GetValue(AUXILIAR_HESSIAN);
+
+        const double nodal_h = it_node->GetValue(NODAL_H);
+
+        double element_min_size = mMinSize;
+        if ((element_min_size > nodal_h) && mEnforceCurrent) element_min_size = nodal_h;
+        double element_max_size = mMaxSize;
+        if ((element_max_size > nodal_h) && mEnforceCurrent) element_max_size = nodal_h;
+
+        // Isotropic by default
+        double ratio = 1.0;
+
+        if (it_node->SolutionStepsDataHas(reference_var)) {
+            const double ratio_reference = it_node->FastGetSolutionStepValue(reference_var);
+            ratio = CalculateAnisotropicRatio(ratio_reference, mAnisotropicRatio, mBoundLayer, mInterpolation);
+        }
+
+        // For postprocess pourposes
+        it_node->SetValue(ANISOTROPIC_RATIO, ratio);
+
+        // We compute the metric
+        KRATOS_DEBUG_ERROR_IF_NOT(it_node->Has(tensor_variable)) << "METRIC_TENSOR_" + std::to_string(TDim) + "D  not defined for node " << it_node->Id() << std::endl;
+        TensorArrayType& metric = it_node->GetValue(tensor_variable);
+
+        const double norm_metric = norm_2(metric);
+        if (norm_metric > 0.0) {// NOTE: This means we combine differents metrics, at the same time means that the metric should be reseted each time
+            const TensorArrayType& old_metric = it_node->GetValue(tensor_variable);
+            const TensorArrayType& new_metric = ComputeHessianMetricTensor<TDim>(hessian, ratio, element_min_size, element_max_size);
+
+            metric = MetricsMathUtils<TDim>::IntersectMetrics(old_metric, new_metric);
+        } else {
+            metric = ComputeHessianMetricTensor<TDim>(hessian, ratio, element_min_size, element_max_size);
+        }
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+Parameters ComputeHessianSolMetricProcess::GetDefaultParameters()
+{
+    Parameters default_parameters = Parameters(R"(
+    {
+        "minimal_size"                        : 0.1,
+        "maximal_size"                        : 10.0,
+        "enforce_current"                     : true,
+        "hessian_strategy_parameters":
+        {
+            "metric_variable"                  : ["DISTANCE"],
```

Comment:
```suggestion
             "metric_variables"                  : ["DISTANCE"],
```
it is a list

Reply:
I will keep it. It is used in many places, and will break the cod of the current users as @AlejandroCornejo